pkg/subctl/cmd/gather/logs.go

```diff
@@ -62,41 +61,45 @@ func gatherPodLogsByContainer(podLabelSelector, container string, info Info) {
 	}
 }
 
-func podLogsToFile(pod *corev1.Pod, podLogOptions *corev1.PodLogOptions, info Info) error {
-	logRequest := info.ClientSet.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, podLogOptions)
+func outputPodLogs(pod *corev1.Pod, podLogOptions corev1.PodLogOptions, info Info) error {
+	err := outputPreviousPodLog(pod, podLogOptions, info)
+	if err != nil {
+		return err
```

Comment:
The problem here is that if this one fails the next one won’t even be attempted. I would save the error and return an aggregated error at the end of the function so we don’t skip the other log dumper.

Reply:
yes. Good catch @mangelajo